extension/surveys/survey-generator.js

```diff
@@ -571,3 +571,45 @@ function getFormValues(questionArr, form) {
   }
   return responses;
 }
+
+/**
+ * Checks whether a form has any responses or text in it.
+ * @param {object} form The form to check.
+ * @returns {boolean} True if the form has content.
+ */
+function formHasContent(form) {
```

Comment:
You could also mark the inputs with class names in HTML (e.g. "checkable-input" for checkbox and radio, "editable-input" for others) and check them here with two loops with no switch statement. But this is probably simpler so lgtm.